ironfish/src/account/accounts.ts

```diff
@@ -921,6 +961,125 @@ export class Accounts {
     await this.scanTransactions()
   }
 
+  getTransactionsFor(account: Account): {
+    transactions: {
+      status: string
+      hash: string
+      isMinersFee: boolean
+      fee: number
+      notes: number
+      spends: number
+    }[]
+  } {
+    this.assertHasAccount(account)
+
+    const transactions = []
+
+    for (const transactionMapValue of this.transactionMap.values()) {
+      const transaction = transactionMapValue.transaction
+
+      // check if transaction creator
+      let transactionCreator = false
+      for (const note of transaction.notes()) {
+        if (note.decryptNoteForSpender(account.outgoingViewKey)) {
+          transactionCreator = true
+          break
+        }
+      }
+
+      if (transactionCreator) {
+        transactions.push({
+          status:
+            transactionMapValue.blockHash && transactionMapValue.submittedSequence
+              ? 'completed'
+              : 'pending',
+          hash: transaction.hash().toString('hex'),
+          isMinersFee: transaction.isMinersFee(),
+          fee: Number(transaction.fee()),
+          notes: transaction.notesLength(),
+          spends: transaction.spendsLength(),
+        })
+      }
+    }
+
+    return { transactions }
+  }
+
+  getTransaction(
+    account: Account,
+    hash: string,
+  ): {
+    transactionInfo: {
+      status: string
+      isMinersFee: boolean
+      fee: number
+      notes: number
+      spends: number
+    } | null
+    transactionNotes: {
+      spender: boolean
+      amount: number
+      memo: string
+    }[]
+  } {
+    this.assertHasAccount(account)
+
+    let transactionInfo = null
+    const transactionNotes = []
+
+    for (const transactionMapValue of this.transactionMap.values()) {
+      const transaction = transactionMapValue.transaction
+
+      if (transaction.hash().toString('hex') === hash) {
+        // check if transaction creator
+        let transactionCreator = false
+        for (const note of transaction.notes()) {
+          if (note.decryptNoteForSpender(account.outgoingViewKey)) {
+            transactionCreator = true
+            break
+          }
+        }
+
+        if (transactionCreator) {
```

Comment:
Same thing here about the `transactionCreator` thing, not able to look up a transaction that a note references if I only received in that transaction

![image](https://user-images.githubusercontent.com/97762857/172954975-41fbc895-61f8-4bfb-a87e-5f4502a4edc2.png)

Reply:
yup, removed this.